### PR TITLE
League/Commonmark update to 1.6.5

### PR DESCRIPTION
**Pull request by xmorave2**
This PR updates league/commonmark dependency to version 1.6.5, updates MarkdownFactory according to https://commonmark.thephpleague.com/1.6/upgrading/ and refactors setting of extensions to be more flexible by removing hardcoded GithubFlavoredMarkdownExtension, but make the parts of github extension a default set.

### CODE REVIEW

File: config/vufind/markdown.ini [hunk 1]
```diff
@@ -35,8 +35,11 @@
 ;     Attributes, ExternalLink, Footnote, HeadingPermalink, Mention, SmartPunct,
 ;     TableOfContents
 ; More about extensions: https://commonmark.thephpleague.com/1.5/extensions/overview/
```
Comment (demiankatz) [inline config/vufind/markdown.ini]: Should we change the various comments and links that reference 1.5 to reference 1.6 instead?

File: config/vufind/markdown.ini [hunk 2]
```diff
@@ -35,8 +35,11 @@
...
-; Some of them could have a configuration, see sections bellow
-;extensions = Attributes,ExternalLink,Footnote,HeadingPermalink,Mention,SmartPunct,TableOfContents
+; Some of them could have a configuration, see sections bellow. Available extensions
+; are: Attributes, Autolink, DisallowedRawHtml, ExternalLink, Footnote,
+; HeadingPermalink, Mention, SmartPunct, Strikethrough, Table, TableOfContents,
+; TaskList.
+extensions = Autolink,DisallowedRawHtml,Strikethrough,Table,TaskList
```
Comment (demiankatz) [inline config/vufind/markdown.ini:40]: Now that we have a default list of extensions, do user who have customized this setting previously need to make any changes for things to work correctly? I doubt that this is being heavily used, but we should at least put a note in the changelog if some action may be necessary.

Comment (demiankatz) [reply config/vufind/markdown.ini:40]: Also, if the setting is commented out, should we now default to this default set, or should we continue to default to an empty array in the factory code?

**Review by demiankatz — CHANGES_REQUESTED** [review]
Thank you very much for this -- I was delaying upgrading this dependency because I suspected it would require some extra work, so I appreciate your initiative to take care of it before I had a chance to ask for help. :-)

In any case, just a couple of minor questions/suggestions regarding the configuration....

File: config/vufind/markdown.ini (hunk 2 above)
Comment (xmorave2) [reply config/vufind/markdown.ini:40]: A note in changelog would be fine. To continue thing work as before, this set of extensions should be enabled.

File: config/vufind/markdown.ini (hunk 1 above)
Comment (xmorave2) [reply config/vufind/markdown.ini]: Definitely, I updated the links and also fixed the list of extensions, which was partly duplicated...

File: config/vufind/markdown.ini (hunk 2 above)
Comment (xmorave2) [reply config/vufind/markdown.ini:40]: I made these extensions a default set if no extensions are set. To remove the default, and use raw commonmark, there is posibility to set the 'extensions' option to empty string

Comment (demiankatz) [reply config/vufind/markdown.ini:40]: Note added -- thanks!

**Comment by demiankatz** [comment]
@xmorave2, I see that you're still doing a bit of work on this -- please let me know when you're ready for it to be merged; I think we're nearly there!

File: config/vufind/markdown.ini (hunk 2 above)
Comment (demiankatz) [reply config/vufind/markdown.ini:40]: (And I revised my note to indicate that manual action is only needed for users who already customized the extensions setting in markdown.ini -- probably a very small number, if not zero). :-)

**Comment by xmorave2** [comment]
> @xmorave2, I see that you're still doing a bit of work on this -- please let me know when you're ready for it to be merged; I think we're nearly there!

Thanks @demiankatz, I am done now, it is ready for your final look ;)

**Comment by demiankatz** [comment]
Looks good to me -- thanks again!

**Pull request merged**